src/backend/metal/src/command.rs

```diff
@@ -938,11 +921,12 @@ fn exec_render<'a>(encoder: &metal::RenderCommandEncoderRef, command: soft::Rend
             encoder.set_stencil_front_back_reference_value(front, back);
         }
         Cmd::BindBuffer { stage, index, buffer, offset } => {
+            let native = buffer.as_ref().map(|b| b.as_native());
```

Comment:
@kvark without `as_ref`, the borrow checker complains because `b` is dropped in these cases